packages/source-map/lib/sourceMap.ts

```diff
@@ -16,12 +16,12 @@ interface MappingMemo {
 	mappings: Set<Mapping>[];
 }
 
-export class SourceMap<Data = unknown> {
+export class SourceMap<Data = unknown, T = unknown> {
 
 	private sourceCodeOffsetsMemo: MappingMemo | undefined;
 	private generatedCodeOffsetsMemo: MappingMemo | undefined;
 
-	constructor(public readonly mappings: Mapping<Data>[]) { }
+	constructor(public readonly mappings: ({ source?: T } & Mapping<Data>)[], private scriptId?: T) { }
```

Comment:
We need SourceMap of the generated file to be able to map from generated offset to any of the local and associated offsets